edit.ml

```diff
@@ -110,61 +110,16 @@ let split_edge instrs pc label pc' =
   new_instrs, pc_map
 
 let replace_uses_in_instruction old_name new_name instr : instruction =
```

Comment:
Returning the map object: sure, can do. Pushing down: I would propose to keep in Edit the transformations that are generic enough to be potentially of use to different analyses/transformations, and push down the optimization-specific ones. If I can reword this one to not have Assign a special case anymore, it becomes generic enough to stay -- otherwise it goes. On the other hand, `freshen_assign` should probably be pushed down in any case.